src/DDTrace/Integrations/Predis.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace DDTrace\Integrations;
+
+use DDTrace\Tags;
+use DDTrace\Types;
+use OpenTracing\GlobalTracer;
+use Predis\Client;
+
+const VALUE_PLACEHOLDER = "?";
+const VALUE_MAX_LEN = 100;
+const VALUE_TOO_LONG_MARK = "...";
+const CMD_MAX_LEN = 1000;
+
+class Predis
+{
+    /**
+     * Static method to add instrumentation to the Predis library
+     */
+    public static function load()
+    {
+        if (!extension_loaded('ddtrace')) {
+            trigger_error('ddtrace extension required to load Predis integration.', E_USER_WARNING);
+            return;
+        }
+
+        // public mixed Predis\Client::executeCommand(CommandInterface $command)
+        dd_trace(Client::class, 'executeCommand', function ($command) {
+            $query = Predis::formatCommandArgs($command);
+            $scope = GlobalTracer::get()->startActiveSpan('Predis.executeCommand');
+            $span = $scope->getSpan();
+            $span->setTag(Tags\SPAN_TYPE, Types\REDIS);
+            $span->setTag(Tags\SERVICE_NAME, 'redis');
+            $span->setTag('redis.raw_command', $query);
+            $span->setTag('redis.args_length', count($command->getArguments()));
+            $span->setResource($query);
```

Comment:
Same question as on sanitizing, my naive understanding was that the agent takes care of this by quantizing, like with SQL queries. If that's not the case what do we want the resource to be here, just the actual command name?

Reply:
Oh, yeah. Sorry for confusion, in Ruby a lot of those were still quantized manually. Looking at the code of https://github.com/DataDog/datadog-trace-agent/blob/master/obfuscate/redis.go
looks like as long as type is set correctly it will be quantized/obfuscated.